src/game/object_list_processor.c

```diff
@@ -486,8 +467,8 @@ void spawn_objects_from_info(UNUSED s32 unused, struct SpawnInfo *spawnInfo) {
             object->unused1 = 0;
 
             // Record death/collection in the SpawnInfo
-            object->respawnInfoType = RESPAWN_INFO_TYPE_NORMAL;
-            object->respawnInfo = &spawnInfo->behaviorArg;
+            object->respawnInfo = spawnInfo->respawnInfo;
+            object->respawnInfoPointer = &spawnInfo->respawnInfo;
```

Comment:
What's the difference between these exactly?

Reply:
reworked respawning to use its own field instead of bparam3 so idk what you mean

Reply:
Where is `respawnInfoPointer` accessed where `respawnInfo` is insufficient?

Reply:
the object needs to know where its respawn info is located so it can write to it, the way area persistence works is that across area changes, objects are reloaded but spawninfo structs aren't. so to mark an object as "gone" the object needs to have a reference to its spawninfo's respawn data so that it can write to it